### DISCUSSION TITLE
Probable noob mistake with simple query

### DISCUSSION THREAD

**Discussion by adamzerner** [supabase/supabase#12010] [Questions] [answered]
For some reason I can't get this query to work. I'm probably making some sort of noob mistake but can't figure it out. I have spent time looking through the [docs](https://supabase.com/docs/reference/javascript/using-filters) and I feel like I'm doing the same things.

```
import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs";
import { NextApiRequest, NextApiResponse } from "next";

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  try {
    const supabase = createServerSupabaseClient({ req, res });
    const response = await supabase
      .from("customers")
      .select()
      .eq("stripe_customer_id", "cus_NFqeiuo0dGZx1A");

    res.status(200).json(response);
  } catch (e: any) {
    res.status(500).json(e);
  }
};

export default handler;
```

I get this response:
```
{
  "error": null,
  "data": [],
  "count": null,
  "status": 200,
  "statusText": "OK"
}
```

`select('id')` and `select('stripe_customer_id')` also give the same response.

But the table and record definitely both exist (I hit refresh):
<img width="1440" alt="Screenshot 2023-01-28 at 12 36 53 PM" src="https://user-images.githubusercontent.com/3144254/215290047-8f435b2d-29f8-4a24-b0f4-215eefe7d055.png">

When I console log `supabase` I see that it exists, there's no problems, and the `schema` is `'public'`, which is what `customers` is under.

**Comment by ruggi99** [comment, marked as answer]
@adamzerner from what I see from the screenshot, you have enabled RLS on the table but haven't defined any policy to read or write data.
Try to create a new policy for `SELECT` and see if it works

**Reply by adamzerner** [reply]
Oh yeah that was it. Thanks.
